src/components/homePage.tsx

```diff
@@ -17,64 +27,142 @@ const CardTitle = ({ title, icon }: { title: string; icon: any }) => {
-        <span className="text-[10px] text-[#F16476]">Difference</span>
+        <span
+          className={`${
+            difference > 0
+              ? "text-[#F16476]"
+              : difference < 0
+              ? "text-[#5CB7ED]"
+              : "text-[#8B8C8F]"
+          } text-[10px]`}
+        >
+          Difference
+        </span>
         <div className="flex">
-          <span className="text-[#F2F2F2] text-[#F16476] text-2xl">38</span>
-          <CustomImage className="flex ml-2" src={trending_down} />
+          <span
+            className={`${
+              difference > 0
+                ? "text-[#F16476]"
+                : difference < 0
+                ? "text-[#5CB7ED]"
+                : "text-[#F2F2F2]"
+            } text-[#F16476] text-2xl`}
+          >
+            {!isNaN(difference) && difference}
+          </span>
+          {difference > 0 ? (
+            <CustomImage className="flex ml-2" src={trending_down} />
+          ) : difference < 0 ? (
+            <CustomImage className="flex ml-2" src={trending_down} />
+          ) : null}
         </div>
       </div>
     </div>
   );
 };
 
 const HomePage = () => {
+  const [canShow, setCanShow] = useState(true);
+  const [avgOPM, setAvgOPM] = useState<number>(0);
+  const [totalOPM, setTotalOPM] = useState<number>(0);
+  const [lastMinOPM, setLastMinOPM] = useState<number>(0);
+  const [lastDayAvgOPM, setLastDayAvgOPM] = useState<number>(0);
+  const [lastDaytotalOPM, setLastDayTotalOPM] = useState<number>(0);
+  const [refreshTime, setRefreshTime] = useState<number>(0);
+
+  const { width } = useScreenSize();
+
+  const fetchOPMData = async (url) => {
+    const opmData = await fetchData(
+      `${url}?period=${ORDERS_DURATION}&starttime=&channel=&promocode=&paymentType=&country=`,
+      {}
+    );
+    const totalOrders = opmData.reduce(
+      (acc, obj) => acc + parseInt(obj.orderCount),
+      0
+    );
+    setTotalOPM(totalOrders);
+    setAvgOPM(Math.round(totalOrders / ORDERS_DURATION));
+    setLastMinOPM(opmData[opmData.length - 1]["orderCount"]);
+  };
+
+  const fetchCompData = async (url, date) => {
+    const opmData = await fetchData(
+      `${url}?period=${ORDERS_DURATION}&date=${date}&channel=&promocode=&paymentType=&country=`,
+      {}
+    );
+    const totalOrders = opmData.reduce(
+      (acc, obj) => acc + parseInt(obj.orderCount),
+      0
+    );
+    setLastDayTotalOPM(totalOrders);
+    setLastDayAvgOPM(Math.round(totalOrders / ORDERS_DURATION));
+  };
+
+  const getCardsData = () => {
+    fetchOPMData(URL_OPM);
+    const date = new Date();
+    date.setDate(date.getDate() - 1);
+    fetchCompData(URL_OPM, date);
+    setRefreshTime(new Date().getTime());
+  };
+
+  useEffect(() => {
+    getCardsData();
+  }, []);
+
   const handleRefreshBtnClick = () => {
-    // fetch api for cards data
+    getCardsData();
   };
 
+  useEffect(() => {
+    const timer = setTimeout(() => setCanShow(!canShow), 30000);
```

Comment:
Keep `30000` in constants so as to configure easily in future.
My guess is it might have to be moved to env variable later.